New in Chrome 103 post (#3058)
GoogleChrome/developer.chrome.com: site/en/blog/new-in-chrome-103/index.md

---
title: New in Chrome 103
description: >

>   Chrome 103 is rolling out now! There's a new HTTP status code that helps the browser decide what content to preload before the page has even started to arrive. The Local Font Access API gives web applications the ability to enumerate and use fonts installed on the user's computer. There's an easier way to implement timeouts on asynchronous APIs. And there's plenty more!

Reviewer: ```suggestion
  Chrome 103 is rolling out now! There's a new HTTP status code that helps the browser decide what content to preload before the page has even started to arrive. The Local Font Access API gives web applications the ability to enumerate and use fonts installed on the user's computer. There's an easier way to implement timeouts on asynchronous APIs. And there's plenty more.
```

* [`AbortSignal.timeout`](#abort-timeout) is an easier way to implement
  timeouts on asynchronous APIs.
* And there's plenty [more](#more).

I'm [Pete LePage](https://petelepage.com). Let's dive in and
see what's new for developers in Chrome 103.

## HTTP 103 status code 103 - early hints {: #http103 }

One way you can improve page performance is to use resource hints. They give
the browser "hints" about what stuff it might need later. For example,
preloading files, or connecting to a different server.

```html
<link as="font" crossorigin="anonymous"
      href="..." rel="preload">
<link as="font" crossorigin="anonymous"
      href="..." rel="preload">
<link href="https://web-dev.imgix.net"
      rel="preconnect">
```

But the browser can't act on those hints until the server sends at least
part of the page.

Imagine the browser requests a page, but the server requires a few hundred
milliseconds to generate it. Until the browser starts to receive the page,
it just sits there and waits. But, if the server knows the page will always
need a certain set of sub-resources, for example, a CSS file, some JavaScript,
and a few images, it can immediately respond with the new HTTP 103

> Early Hints status code, and ask the browser to preload those sub-resources.

Reviewer: ```suggestion
Early Hints status code, and ask the browser to preload those subresources.
```

Then, once the server has generated the page, it can send it with the normal
HTTP 200 response. As the page comes in, the browser has already started
loading the required resources.

Since this is a new HTTP status code, using it requires updates to your server.

Get started with HTTP 103 Early hints:

* [Explainer for Early Hints](https://github.com/bashi/early-hints-explainer/blob/main/explainer.md)
* [Apache 2 Early Hints configuration](https://httpd.apache.org/docs/2.4/howto/http2.html#earlyhints)
* [Using Early Hints on Cloudflare](https://developers.cloudflare.com/cache/about/early-hints/)
* [Fastly Beyond Server Push: The 103 Early Hints Status Code](https://www.fastly.com/blog/beyond-server-push-experimenting-with-the-103-early-hints-status-code)

> ## Local font access API {: #local-fonts }

Reviewer: ```suggestion
## Local Font Access API {: #local-fonts }
```

Fonts on the web have always been a challenge, and especially so for apps
that let users create their own graphics and designs. Until now, web apps
could only really use web fonts.

There was no way to get a list of fonts the user had installed on their
computer. And, there was no way to access the full font table data, critical
if you need to implement your own custom text stack.

The new Local Font Access API gives web applications the ability to enumerate

> the local fonts on the users device, and provides access to the font table data.

Reviewer: ```suggestion
the local fonts on the user's device, and provides access to the font table data.
```
